v2/internal/resolver/resource_hierarchy.go

```diff
@@ -151,14 +176,82 @@ func (h ResourceHierarchy) fullyQualifiedARMIDImpl(subscriptionID string, origin
 		// Join them together
 		interleaved := genruntime.InterleaveStrSlice(resourceTypes, azureNames)
 		return genericarmclient.MakeTenantScopeARMID(provider, interleaved...)
+	case ResourceHierarchyRootARMID:
+		// TODO: Possibly refactor this huge method into sub-functions?
```

Comment:
Good idea.

Reply:
In the interest of time I'll file an item for this and do it later.

Reply:
See #3254